### PR TITLE
wrapping C++ struct

### PR DESCRIPTION
**Pull request by sunayana**

### CODE REVIEW

**Review by croissant-tmp — APPROVED** [review]
Amazing work!
I was able to successfully reproduce in a container all the steps outlined in the ShortCut story 38122.

File: wrappers/RichDEM.jl/src/RichDEM.jl [hunk 1]
```diff
@@ -25,14 +25,42 @@ end #misc
 
 module depressions
     using CxxWrap
+    mutable struct jlDepression{T}
+        pit_cell::UInt32
```
Comment (sunayana) [inline wrappers/RichDEM.jl/src/RichDEM.jl:29]: change `jlDepression` to `Depression`

File: wrappers/RichDEM.jl/src/RichDEM.jl [hunk 2]
```diff
@@ -25,14 +25,42 @@ end #misc
...
     function __init__()
         @initcxx
     end
 
-    # Base.size(dep_hier::DepressionHierarchyDouble) = size(dep_hier)
-    # Base.IndexStyle(::Type{<:DepressionHierarchyDouble}) = IndexCartesian()
+    
+    
+    jl_ocean_linked(dep::Depression) = ocean_linked(dep)
+    function make_jl_depression(dep::Depression)
```
Comment (sunayana) [inline wrappers/RichDEM.jl/src/RichDEM.jl:54]: Outer constructor for jlDepression

File: src/jlrichdem/depressions/jldepressions.cpp [hunk 3]
```diff
@@ -8,41 +9,119 @@
...
+                           { return vec.size(); });
 
-    mod.method("GetDepressionHierarchyDoubleD8", [](const rd::Array2D<double> &topo,
-                                                    rd::Array2D<rd::dephier::dh_label_t> &label,
-                                                    rd::Array2D<int8_t> &flowdirs)
-               { return rd::dephier::GetDepressionHierarchy<double, rd::Topology::D8>(topo, label, flowdirs); });
+            wrapped.module().set_override_module(jlcxx::stl::StlWrappers::instance().module());
+            wrapped.method("cxxgetindex", [](const WrappedT &v, int_t i) -> typename WrappedT::const_reference
+                           { return v[i - 1]; });
+            wrapped.method("cxxgetindex", [](WrappedT &v, int_t i) -> typename WrappedT::reference
```
Comment (Libbum) [inline src/jlrichdem/depressions/jldepressions.cpp:71]: What is the need for `reference` and `const_reference`?

**Review by Libbum — APPROVED** [review]
Think we're good with method and implementation for the moment. We should spend some time writing up the design choices and tech debt (e.g. `DepressionSubset -> Depression`) for the release.

**Pull request merged**